### PR TITLE
Caculated MD5 is incorrect

### PR DESCRIPTION
**Pull request by legege**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ function error(msg) {
 function hash(file, options) {
 	return typeof options.algo === 'function'
 		? options.algo.call(undefined, file)
-		: crypto.createHash(options.algo).update(file.contents.toString()).digest('hex');
+		: crypto.createHash(options.algo).update(file.contents, 'binary').digest('hex');
```
Comment (UltCombo) [inline index.js]: Mind removing the `'binary'` argument? From [crypto docs](https://nodejs.org/api/crypto.html#crypto_cipher_update_data_input_encoding_output_encoding):

> If `data` is a Buffer then `input_encoding` is ignored.

**Comment by UltCombo** [comment]
Thanks for noticing this. I've added some review comments above.
If you don't have time to amend the commit, I can do it later after work.

**Comment by brandonshults** [comment]
It seems like it is agreed that this pull request is a good idea.  I've encountered a situation where it is pretty crucial to a project I am working on.  Is there any chance we can get it added into master with the amended changes?

I'm willing to create a new pull request with the fixes you suggested if that would be preferred.

**Comment by UltCombo** [comment]
Oh my bad, this fell out of my radar. I think I should be able to pull this in and publish a patch release tonight. I also have to write a changelog entry for the new release.

I'm also considering updating our stream library to `through2` which should improve the overall performance, as well as updating the other outdated dependencies, but I guess this can be done in a future release after this patch.

**Comment by brandonshults** [comment]
Thank you for the quick response, this will be a great help.

**Comment by UltCombo** [comment]
Rebased, cleaned up and landed the PR here: ddc09b9972f82992ec5b9e5a2cccb7724d5ca191
Added test here: 855ae782e9aebf62075db18a1310b1b2327b2889
Thanks @legege, and sorry for the delay.

p.s. The tests are working fine in Node@6.0.0, but the new test is broken on Node 0.10 which does not support `Buffer.from`. Guess I can switch to the deprecated [Buffer constructor overload](https://nodejs.org/api/buffer.html#buffer_new_buffer_array) for backwards compatibility.

**Comment by UltCombo** [comment]
**Published v1.0.1!** :tada: :confetti_ball:

It includes this fix and official support for Node 0.12, 4, 5 and 6. Changelog available [here](https://github.com/UltCombo/gulp-buster/blob/master/CHANGELOG.md#patch-releases).

**Comment by UltCombo** [comment]
Aaaand v1.0.2 released with updated dependencies! :zap:

**Pull request closed without merging**